rs/consensus/src/consensus/validator.rs

```diff
@@ -943,7 +1012,7 @@ impl Validator {
                         // A successful verification is enough to validate this block,
                         // because from the notarization we know that the block validity
                         // was already checked.
-                        known_qualified_ranks.insert(proposal.height(), Some(proposal.rank()));
+                        valid_qualified_ranks.add(&proposal);
```

Comment:
It's a corner case but I guess it could happen that `disqualified_ranks` already contains `proposal.rank()` (if we have a notarization and a equiovaction proof for the same block).

Is it a problem if `valid_qualified_ranks` and `disqualified_ranks` are not disjoint?

Reply:
Thinking about it, I think it makes sense if we keep them disjoint. So we only add to `valid_ranks` a notarized proposal, if it has not been DQd before. 

This could theoretically cause some more validations then necessary - because we don't consider notarized equivocating blocks when we skip higher ranks -  but that's anyways an optimization, and doesn't seem strictly necessary. 

I've changed it to only include the proposal in `valid_ranks`, if it's not DQd.